### PR TITLE
[DO NOT MERGE UNTIL PG16 GA] updating postgres major version to 16

### PR DESCRIPTION
**Pull request by brittanyrjones**
updating to major postgres version 16
GUS: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00001mYgo3YAC/view
pg16 Release notes: https://www.postgresql.org/docs/release/16.0/

Also removes EOL build images and PG versions

### CODE REVIEW

File: bin/compile [hunk 1]
```diff
@@ -18,13 +18,14 @@ BUILDPACK_DIR="$(dirname $(dirname $0))"
...
 S3_BUCKET="ci-database-binary"
-POSTGRESQL_MAJOR_VERSION=15
+POSTGRESQL_MAJOR_VERSION=16
 if [ -f "$ENV_DIR/POSTGRESQL_VERSION" ]
 then
   POSTGRESQL_MAJOR_VERSION="$(cat $ENV_DIR/POSTGRESQL_VERSION)"
 fi
 
 case "$POSTGRESQL_MAJOR_VERSION" in
+  16)  POSTGRESQL_VERSION="16.1";;
```
Comment (valeriecodes) [inline bin/compile:28]:
```suggestion
  16)  POSTGRESQL_VERSION="16.2";;
```

File: support/pg_versions [hunk 2]
```diff
@@ -1 +1 @@
-9.6.24;10.22;11.17;12.12;13.8;14.5;15.2
+9.6.24;10.22;11.17;12.12;13.8;14.5;15.2;16.1
```
Comment (valeriecodes) [inline support/pg_versions:1]:
```suggestion
9.6.24;10.22;11.17;12.12;13.8;14.5;15.2;16.2
```

**Review by valeriecodes — APPROVED** [review]
LGTM with requested changes

**Review by valeriecodes — COMMENTED** [review]
Seems like last year we waited for GA, should we do the same? https://github.com/heroku/heroku-buildpack-ci-postgresql/pull/61

**Comment by brittanyrjones** [comment]
Yes that makes sense to me will not merge until we GA.

**Review by kbarber — APPROVED** [review]

**Pull request merged**